### PR TITLE
Refactor hardhat to localnode

### PR DESCRIPTION
**Pull request by evzpav**

### CODE REVIEW

File: localnode/localnode.go [hunk 1]
```diff
@@ -0,0 +1,359 @@
...
+	nodeURL     string
+	cmd         *exec.Cmd
+	accounts    []tx_p.Account
+	logger      log.Logger
+}
+
+func New(logger log.Logger, nodeType NodeType, forkNodeURL string, blockNumber string) *localNode {
+	if forkNodeURL == "" {
+		logger.Log("invalid forkNodeURL")
+		os.Exit(1)
```
Comment (krasi-georgiev) [inline localnode/localnode.go]: Where possible lets return an error and handle it in the caller. Here and few other places.
I usually use `panic` in cases where the error happens in a goroutine and returning an error would complicate the code too much

File: localnode/localnode.go [hunk 2]
```diff
@@ -0,0 +1,359 @@
...
+		if err != nil {
+			panic(err)
+		}
+		Accounts = append(Accounts, acc)
+	}
+
+	return Accounts
+}
+
+type localNode struct {
```
Comment (krasi-georgiev) [inline localnode/localnode.go:63]: I don't think you can call public methods on a private struct.

File: localnode/localnode.go (hunk 1 above)
Comment (evzpav) [reply localnode/localnode.go]: Agreed. I was following old pattern here... I'll change

File: localnode/localnode.go (hunk 2 above)
Comment (evzpav) [reply localnode/localnode.go:63]: you can.... I've been using this pattern for a while... works for sure

File: localnode/localnode.go [hunk 3]
```diff
@@ -69,10 +70,9 @@ type localNode struct {
 	logger      log.Logger
 }
 
-func New(logger log.Logger, nodeType NodeType, forkNodeURL string, blockNumber string) *localNode {
+func New(logger log.Logger, nodeType NodeType, forkNodeURL string, blockNumber string) (*localNode, error) {
 	if forkNodeURL == "" {
-		logger.Log("invalid forkNodeURL")
-		os.Exit(1)
+		return nil, fmt.Errorf("invalid forkNodeURL")
```
Comment (krasi-georgiev) [inline localnode/localnode.go]: should use the errors package to be consistent
`errors.Errorf(`

**Pull request merged**